WordPress/src/main/java/org/wordpress/android/viewmodel/PluginBrowserViewModel.kt

```diff
@@ -0,0 +1,465 @@
+package org.wordpress.android.viewmodel
+
+import android.arch.lifecycle.LiveData
+import android.arch.lifecycle.MutableLiveData
+import android.arch.lifecycle.ViewModel
+import android.os.Bundle
+import android.os.Handler
+import android.support.annotation.WorkerThread
+import android.text.TextUtils
+import org.greenrobot.eventbus.Subscribe
+import org.greenrobot.eventbus.ThreadMode
+import org.wordpress.android.WordPress
+import org.wordpress.android.fluxc.Dispatcher
+import org.wordpress.android.fluxc.generated.PluginActionBuilder
+import org.wordpress.android.fluxc.model.SiteModel
+import org.wordpress.android.fluxc.model.plugin.ImmutablePluginModel
+import org.wordpress.android.fluxc.model.plugin.PluginDirectoryType
+import org.wordpress.android.fluxc.store.PluginStore
+import org.wordpress.android.fluxc.store.PluginStore.FetchPluginDirectoryPayload
+import org.wordpress.android.util.AppLog
+import org.wordpress.android.util.AppLog.T
+import org.wordpress.android.util.StringUtils
+import java.util.ArrayList
+import java.util.HashMap
+import java.util.HashSet
+import javax.inject.Inject
+import kotlin.properties.Delegates
+
+private const val KEY_SEARCH_QUERY = "KEY_SEARCH_QUERY"
+private const val KEY_TITLE = "KEY_TITLE"
+
+@WorkerThread
```

Comment:
I moved the `@WorkerThread` declaration to the class (from having it on methods specifically). I don't think there is any method that has to be in the main thread in this VM and I don't think there should ever be one. As long as we use `postValue` for `LiveData` instead of `setValue` we should be safe.

If we find that something has to be done in a main thread, that might be a sign that either it doesn't belong to the VM or it should be done through `LiveData`.

Reply:
A View Model holds the state of data displayed by the UI. That means it's not necessarily only stuff you need to load from the network or a database. For example, there could be some text that changes based on user interaction, which would not be asynchronous. Yet you still want to observe that change with a `LiveData` and would youse `setValue` in this case.

Reply:
Yes, I'd use `postValue` every time a change is affecting the UI. If for some reason, we need a UI change to happen immediately and we need to call a VM method for it (which seems unlikely to me) then we can mark that method with `@MainThread`. It should be a pretty rare case and I think it should be a conscious decision to do something in main thread for ViewModels.

Reply:
Just wanted to drop another note that, I don't expect handling every method in `@WorkerThread` to make a big difference in most cases. However, for a feature like plugins where 5 different lists are managed and updated from multiple sources, it can actually make a noticeable difference (at least I expect it to). I still need to learn how to profile these and see the difference in numbers.

I also think it's a good practice to handle everything non-ui related in a different thread.